src/main/java/io/supertokens/utils/EmailValidator.java

```diff
@@ -0,0 +1,34 @@
+/*
+ *    Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ *    This software is licensed under the Apache License, Version 2.0 (the
+ *    "License") as published by the Apache Software Foundation.
+ *
+ *    You may not use this file except in compliance with the License. You may
+ *    obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.utils;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public abstract class EmailValidator {
```

Comment:
You don't need a new class for this.. It can simply be a static function in the `utils.java` class.. No need to create extra classes please.

Reply:
Just as we discussed, `Utils.java` is bad. Everything is mixed there and it's harder to find various utilities (since they are namespaced). I think we will be better with `EmailValidator.java`: no effort to make and test it, higher clarity and no dependency magnets (as `Utils.java` is).

Reply:
I would argue that having a class for just email validation is a bit too much.. a class called `Validators` which contains logic for email validation and other types of validation makes sense though.

The reason I think it's a bit too much is cause over time, we may have tons of validator classes, each containing just one function... But OK - I can accept this, as we really need to move faster.